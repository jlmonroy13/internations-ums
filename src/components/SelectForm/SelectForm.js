import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SelectForm extends Component {
  constructor() {
    super();
    this.state = {
      itemId: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleSubmit(event) {
    event.preventDefault();
    const { itemId } = this.state;
    const { type, urlId } = this.props;
    if (type === 'user') {
      this.props.addGroupToUser(urlId, itemId);
    } else {
      this.props.addUserToGroup(urlId, itemId);
    }
  }
  handleChange(event) {
    const { name, value } = event.target;
    this.setState(() => ({
      [name]: value,
    }));
  }

  render() {
    const { type, items } = this.props;
    const className = type === 'user' ? 'group' : 'user';
    function onRenderOptions(item) {
      return (
        <option value={item.id} key={item.id}>
          {className === 'group' ? item.name : `${item.firstName} ${item.lastName}`}
        </option>
      );
    }
    return (
      <form onSubmit={this.handleSubmit} className="list__select-form">
        <label htmlFor="items" className="form__label" />
        <div className="list__select-container">
          <select className="list__select" id="itemId" name="itemId" onChange={this.handleChange}>
            <option value="">{`Add ${className}`}</option>
            {items.map(onRenderOptions)}
          </select>
        </div>
        <button type="submit" className="list__select-button" disabled={!this.state.itemId} >
          <span className={`list__button icon icon--add-${className}`} />
        </button>
      </form>
    );
  }
}

SelectForm.propTypes = {
  type: PropTypes.string.isRequired,
  urlId: PropTypes.string.isRequired,
  items: PropTypes.array.isRequired,
  addGroupToUser: PropTypes.func.isRequired,
  addUserToGroup: PropTypes.func.isRequired,
};

export default SelectForm;
