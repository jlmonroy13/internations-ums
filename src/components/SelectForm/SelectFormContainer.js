import { connect } from 'react-redux';
import { addGroupToUser } from '../../actions/user';
import { addUserToGroup } from '../../actions/group';
import SelectForm from './SelectForm';

function mapDispatchToProps(dispatch) {
  return {
    addGroupToUser: (contactId, groupId) => dispatch(addGroupToUser(contactId, groupId)),
    addUserToGroup: (groupId, contactId) => dispatch(addUserToGroup(groupId, contactId)),
  };
}

export default connect(null, mapDispatchToProps)(SelectForm);
