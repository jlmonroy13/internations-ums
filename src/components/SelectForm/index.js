import SelectFormContainer from './SelectFormContainer';

export default SelectFormContainer;
