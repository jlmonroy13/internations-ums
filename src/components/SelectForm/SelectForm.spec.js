import React from 'react';
import renderer from 'react-test-renderer';
import SelectForm from './SelectForm';

describe('<SelectForm />', () => {
  it('should render the select form component', () => {
    const addGroupToUser = jest.fn();
    const addUserToGroup = jest.fn();
    const wrapper = renderer.create(
      <SelectForm
        addGroupToUser={addGroupToUser}
        addUserToGroup={addUserToGroup}
        type=""
        urlId=""
        items={[]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
