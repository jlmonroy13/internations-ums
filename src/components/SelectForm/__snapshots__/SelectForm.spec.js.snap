// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectForm /> should render the select form component 1`] = `
<form
  className="list__select-form"
  onSubmit={[Function]}
>
  <label
    className="form__label"
    htmlFor="items"
  />
  <div
    className="list__select-container"
  >
    <select
      className="list__select"
      id="itemId"
      name="itemId"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Add user
      </option>
    </select>
  </div>
  <button
    className="list__select-button"
    disabled={true}
    type="submit"
  >
    <span
      className="list__button icon icon--add-user"
    />
  </button>
</form>
`;
